api/tests/routes/pro/post_stocks_test.py

```diff
@@ -564,7 +564,14 @@ def when_missing_offer_id(self, client):
 
         # Then
         assert response.status_code == 400
-        assert response.json == {"offerId": ["Ce champ est obligatoire"]}
+        assert response.json == {
+            "offerId": ["Ce champ est obligatoire"],
+            "stocks.0.id": ["Ce champ est obligatoire"],
+            "stocks.0.quantity": [
+                "Saisissez un nombre supérieur ou égal à 1",
+                "Saisissez un nombre supérieur ou égal à 1",
```

Comment:
C'est bizarre qu'on ait une liste avec 2 fois la même erreur non ?

Reply:
C'est parce que le stock peut être 2 objets différents, du coup la validation passe par les 2

Reply:
Ah ouais j'avais jamais vu le : 
```
"stocks.0.price": [
           "La valeur n'est pas un nombre décimal valide",
           "La valeur n'est pas un nombre décimal valide",
]
```

C'est moche :o

Reply:
Je pense qu'on peut améliorer le truc avec un 
`discriminator="id"` 

> :param discriminator: only useful with a (discriminated a.k.a. tagged) `Union` of sub models with a common field.
      The `discriminator` is the name of this common field to shorten validation and improve generated schema

un peu comme https://github.com/pass-culture/pass-culture-main/blob/e30bc832b7a3399603f81e65282cf9561a65c992/api/src/pcapi/routes/public/individual_offers/v1/serialization.py#L153 . 
Tu peux essayer avec si tu arrives à fixer ça (sans trop y passer trop de temps)

Reply:
Ah mais en faite on a pas d'id dans la création. Je ne sais pas si un `id: None` dans la création permettrait de faire fonctionner le discriminator. 
Sinon tantpis